WcaOnRails/app/controllers/delegates_controller.rb

```diff
@@ -7,4 +7,56 @@ class DelegatesController < ApplicationController
   def stats
     @delegates = User.delegates.includes(:actually_delegated_competitions)
   end
+
+  def probations
+    @probation_roles = Role.where(group_id: UserGroup.where(group_type: "delegate_probation"))
+    user_ids = @probation_roles.pluck(:user_id)
+    @probation_users = User.find(user_ids).index_by(&:id)
+  end
+
+  def delegate_probation_data
+    respond_to do |format|
+      format.json do
+        @probation_roles = Role.where(group_id: UserGroup.where(group_type: "delegate_probation"))
+        @probation_users = {}
+        @probation_roles.each { |probation_role|
+          @probation_users[probation_role.user_id] = User.find_by_id(probation_role.user_id)
+        }
+        render json: {
+          probationUsers: @probation_users.as_json,
+          probationRoles: @probation_roles.as_json,
+        }
+      end
+    end
+  end
```

Comment:
This complete method is duplicated with `def probations` above. One of them should go, it definitely shouldn't be necessary to compute the same data in two different places,

Reply:
Looking at the rest of the code, the part in `def probations` seems to be the one that is unnecessary.
Use the short, concise syntax from there and put it under `delegate_probation_data` instead.